### ISSUE TITLE
cuda.sparse_spmv test failure on Kepler arch with cusparse in Trilinos nightly build

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos-kernels#833]
The `cuda.sparse_spmv_double_int_int_TestExecSpace` began failing after merge of #831 in nightly testing of KokkosKernels in Trilinos in a cuda/9.2 build on Kepler arch. I reproduced in standalone kokkos-kernels with cusparse enabled on kokkos-dev. @brian-kelley are you able to look into this?

Error output:
```bash
22:59:53 3: [ RUN      ] cuda.sparse_spmv_double_int_int_TestExecSpace
22:59:54 3: terminate called after throwing an instance of 'std::runtime_error'
22:59:54 3:   what():  cudaDeviceSynchronize() error( cudaErrorIllegalAddress): an illegal memory access was encountered /home/jenkins/slave/workspace/KokkosKernels_Trilinos_TpetraCore_KokkosDev_CUDA92_opt/Trilinos/kokkos/core/src/Cuda/Kokkos_Cuda_Instance.cpp:143
```

Reproducer instructions:
```bash
# Standalone kokkos-kernels

module load sems-cmake/3.12.2 sems-cuda/9.2 sems-gcc/7.2.0 

KOKKOS_PATH=$HOME/kokkos
KOKKOSKERNELS_PATH=$HOME/kokkos-kernels

../../cm_generate_makefile.bash --with-cuda --with-arch=Kepler35 --compiler=$KOKKOS_PATH/bin/nvcc_wrapper --kokkos-path=$KOKKOS_PATH --kokkoskernels-path=$KOKKOSKERNELS_PATH --with-cuda-options=force_uvm --with-tpls=cusparse


# Nightly Trilinos build
cd $TRILINOS_DIR
ln -snf ${WORKDIR}/kokkos kokkos
ln -snf ${WORKDIR}/kokkos-kernels kokkos-kernels

mkdir -p $BUILDDIR
cd $BUILDDIR

# Configure Trilinos
export ATDM_CONFIG_REGISTER_CUSTOM_CONFIG_DIR=${TRILINOS_DIR}/cmake/std/atdm/contributed/kokkos-dev
source ${TRILINOS_DIR}/cmake/std/atdm/load-env.sh kokkos-dev-cuda-opt

module list
export OMPI_CXX=$KOKKOS_DIR/bin/nvcc_wrapper

cmake \
 -GNinja \
 -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
 -DCMAKE_INSTALL_PREFIX="${PWD}/install" \
 -DCMAKE_CXX_STANDARD="14" \
 -DTrilinos_ENABLE_TESTS=OFF \
 -DTrilinos_ENABLE_ALL_PACKAGES=OFF \
 -DTrilinos_ENABLE_Kokkos=ON \
  -DKokkos_ARCH_KEPLER35=ON \
 -DTrilinos_ENABLE_KokkosKernels=ON \
  -DKokkosKernels_ENABLE_TESTS=ON \
 -DTrilinos_ENABLE_TpetraCore=ON \
   -DTpetra_ENABLE_TESTS=ON \
 -DKokkos_ENABLE_CUDA=ON \
 -DKokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
 -DKokkosKernels_SOURCE_DIR_OVERRIDE:STRING=kokkos-kernels \
$TRILINOS_DIR

# Build Trilinos
make NP=8 VERBOSE=1

# Test install
#make install NP=4

# All Trilinos tests
ctest -V --output-on-failure

# Specific failure
cd kokkos-kernels/unit_test
./KokkosKernels_sparse_cuda.exe --gtest_filter=cuda.sparse_spmv_double_int_int_TestExecSpace

```

**Comment by brian-kelley** [comment]
@ndellingwood I replicated this and I'm looking into it.

**Comment by ndellingwood** [comment]
Thanks @brian-kelley , in case the extra info is helpful the tpl spot-check (cusparse enabled) passed last night on kokkos-dev-2 which tested cuda/10.1 on Volta

**Comment by brian-kelley** [comment]
Yeah I got a crash on kepler but not my pascal workstation.

**Comment by ndellingwood** [comment]
Resolved by PR #834 and nightly build now passing, thanks @brian-kelley !
